### ISSUE TITLE
bitstring library needs to be pinned to 3.1.9 for some executables in ports/espressif/esp-idf/esptool_py to function

### ISSUE THREAD

**Issue by evildave666** [adafruit/circuitpython#7476]
### CircuitPython version

```python
NA
```

### Code/REPL

```python
$ ports/espressif/esp-idf/components/esptool_py/esptool/espefuse.py
```

### Behavior

imports fail due to referencing a class alias that was removed in bitstring library 4.0. The version needs to be forced to 3.1.9 in the build environment.

**Comment by evildave666** [comment]
Should probably be addressed upstream but the building document should mention it at least in the meantime.

**Comment by tannewt** [comment]
I wonder if we should do one more IDF bump before 8.0 RC. This fix is in the upstream 4.4 branch but ours is too old: https://github.com/adafruit/esp-idf/commits/release/v4.4-circuitpython

**Comment by dhalbert** [comment]
> I wonder if we should do one more IDF bump before 8.0 RC. This fix is in the upstream 4.4 branch but ours is too old: https://github.com/adafruit/esp-idf/commits/release/v4.4-circuitpython

I was just going to try merging v4.4 from upstream experimentally to see if it fixed any network or I2C problems. There are a number of interesting commits past where we last merged from.

**Comment by evildave666** [comment]
Fixed by https://github.com/adafruit/circuitpython/pull/7486
